app/scripts/controllers/transactions/index.js

```diff
@@ -2070,8 +2074,8 @@ export default class TransactionController extends EventEmitter {
         : TRANSACTION_ENVELOPE_TYPE_NAMES.LEGACY,
       first_seen: time,
       gas_limit: gasLimit,
-      ...gasParamsInGwei,
       ...extraParams,
+      ...gasParamsInGwei,
```

Comment:
to be safe, may we add a comment here to mention `gasParamsInGwei` needs to come after `...extraParams`?